pkg/mqtt5/client.go

```diff
@@ -161,6 +166,81 @@ func (c *Mqtt5Client) Disconnect() error {
 	}
 
 	c.isConnected = false
+	c.logger.Infof("Disconnected to %s://%s:%s with Client ID: %s", c.configuration.Protocol, c.configuration.Host, strconv.Itoa(c.configuration.Port), c.configuration.ClientID)
+	return nil
+}
+
+// Subscribe creates subscriptions for the specified topics.
+func (c *Mqtt5Client) Subscribe(topics []string, messageChannel chan<- config.MessageEnvelope) error {
+	handler := newDefaultMessageHandler(messageChannel)
+	if err := c.SubscribeWithMessageHandler(topics, handler); err != nil {
+		c.logger.Errorf("Failed to subscribe with default message handler: %v", err)
+		return err
+	}
+	return nil
+}
+
+// SubscribeWithMessageHandler creates subscriptions for the specified topics with the message handler.
+func (c *Mqtt5Client) SubscribeWithMessageHandler(topics []string, messageHandler paho.MessageHandler) error {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	var subscriptions []paho.SubscribeOptions
+	for _, topic := range topics {
+		c.mqtt5Client.Router.RegisterHandler(topic, messageHandler)
+		sub := paho.SubscribeOptions{Topic: topic, QoS: byte(c.configuration.QoS)}
+		subscriptions = append(subscriptions, sub)
+	}
+
+	sa, err := c.mqtt5Client.Subscribe(c.ctx, &paho.Subscribe{
+		Subscriptions: subscriptions,
+	})
+	for _, code := range sa.Reasons {
```

Comment:
add a comment to indicate code == QoS means successful

Reply:
added